File: Plugins/Aspose_Barcode_Java_for_Ruby/lib/asposebarcodejava/BarcodeRecognition/getbarcoderegioninfo.rb
module Asposebarcodejava
  module GetBarcodeRegionInfo
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # initialize barcode reader
        img = data_dir + "barcode.jpg"
        barcode_reader_type = Rjb::import('com.aspose.barcoderecognition.BarCodeReadType')
        reader = Rjb::import('com.aspose.barcoderecognition.BarCodeReader').new(img, barcode_reader_type.Code39Standard)

        # Try to recognize all possible barcodes in the image
        while reader.read()
            # Get the region information
            region = reader.getRegion()
            
            if region != nil
                # Display x and y coordinates of barcode detected
                point = region.getPoints()
                puts "Top left coordinates: X = " + point[0].x.to_s + ", Y = " + point[0].y.to_s
                puts "Bottom left coordinates: X = " + point[1].x.to_s + ", Y = " + point[1].y.to_s
                puts "Bottom right coordinates: X = " + point[2].x.to_s + ", Y = " + point[2].y.to_s
                puts "Top right coordinates: X = " + point[3].x.to_s + ", Y = " + point[3].y.to_s
            end
            puts "Codetext: " + reader.getCodeText().to_s
        end

        # Close reader
        reader.close()
    end
  end
end

File: Plugins/Aspose.BarCode Java for Jython/asposebarcode/WorkingWithBarcodeRecognition/AdvancedBarcodeRecognitionFeatures/GetBarcodeRegionInfo.py
from asposebarcode import Settings
from com.aspose.barcoderecognition import BarCodeReadType
from com.aspose.barcoderecognition import BarCodeReader

class GetBarcodeRegionInfo:

    def __init__(self):
        dataDir = Settings.dataDir + 'WorkingWithBarcodeRecognition/AdvancedBarcodeRecognitionFeatures/GetBarcodeRegionInfo/'
        
        # initialize barcode reader
        img = dataDir + "barcode.jpg"
        barcode_reader_type = BarCodeReadType
        reader = BarCodeReader(img, barcode_reader_type.Code39Standard)

        # Try to recognize all possible barcodes in the image
        while (reader . read()):
        # Get the region information
            region = reader.getRegion()

            if (region != ""):
                # Display x and y coordinates of barcode detected
                point = region.getPoints()
                print "Top left coordinates: X = " 
                print point[0].x 
                print ", Y = " 
                print point[0].y
                print "Bottom left coordinates: X = "
                print point[1].x 
                print ", Y = "
                print point[1].y
                print "Bottom right coordinates: X = " 
                print point[2].x 
                print ", Y = " 
                print point[2].y
                print "Top right coordinates: X = "
                print point[3].x
                print ", Y = " 
                print point[3].y

            print "Codetext: " + reader.getCodeText()
        
        # Close reader
        reader.close()


if __name__ == '__main__':        
    GetBarcodeRegionInfo()